src/data/markdown/docs/01 guides/02 Using k6/14 Executors.md

```diff
@@ -0,0 +1,295 @@
+  scenarios: {
+    constant_arr_rate: {  // arbitrary scenario name
+      executor: 'constant-arrival-rate',
+      rate: 200,
+      timeUnit: '1s',
+      duration: '1m',
+      preAllocatedVUs: 50,
+      maxVUs: 100,
+    }
+  }
+};
+
+export default function() {
+  http.get('https://test.k6.io/');
+}
+```
+
+</div>
+
+
+- Execute a variable RPS test, starting at 50, ramping up to 200 and then back to 0,
+  over a 1 minute period:
+
+<div class="code-group" data-props='{"labels": [ "ramping-arr-rate.js" ], "lineNumbers": "[true]"}'>
+
+```js
+import http from 'k6/http';
+
+export let options = {
+  discardResponseBodies: true,
+  scenarios: {
+    ramping_arrival: {  // arbitrary scenario name
+      executor: 'ramping-arrival-rate',
+      startRate: 50,
+      timeUnit: '1s',
+      preAllocatedVUs: 50,
+      maxVUs: 100,
+      stages: [
+        { target: 200, duration: '30s' },
+        { target: 0, duration: '30s' },
+      ],
+    }
+  }
+};
+
+export default function() {
+  http.get('https://test.k6.io/');
+}
+```
+
+</div>
+
+
+- Run as many iterations as possible with 50 VUs for 30s, and then run 100 iterations
+  per VU of another scenario.
+
+Note the use of `startTime`, and different `exec` functions for each scenario.
+
+<div class="code-group" data-props='{"labels": [ "multiple-scenarios.js" ], "lineNumbers": "[true]"}'>
+
+```js
+import http from 'k6/http';
+
+export let options = {
+  discardResponseBodies: true,
+  scenarios: {
+    const_vus: {
+      executor: 'constant-vus',
+      exec: 'scenario1',
+      vus: 50,
+      duration: '30s',
+    },
+    per_vu_iters: {
+      executor: 'per-vu-iterations',
+      exec: 'scenario2',
+      vus: 50,
+      iterations: 100,
+      startTime: '30s',
+      maxDuration: '1m',
+    },
+  }
+};
+
+export function scenario1() {
+  http.get('https://test.k6.io/', { tags: { scenario: '1' }} );
+}
+
+export function scenario2() {
+  http.get('https://test.k6.io/', { tags: { scenario: '2' }} );
+}
+```
+
+</div>
+
+- Use different environment variables and tags per scenario.
+
+In the previous example we set tags on individual HTTP request metrics, but this
+can also be done per scenario, which would apply them to other
+[taggable](https://k6.io/docs/using-k6/tags-and-groups#tags) objects as well.
+
+<div class="code-group" data-props='{"labels": [ "multiple-scenarios-env-tags.js" ], "lineNumbers": "[true]"}'>
+
+```js
+import http from 'k6/http';
+import { fail } from 'k6';
+
+export let options = {
+  discardResponseBodies: true,
+  scenarios: {
+    const_vus: {
+      executor: 'constant-vus',
+      exec: 'scenario1',
+      vus: 50,
+      duration: '30s',
+      tags: { scenario: '1' },
+      env: { SCENARIO: '1' },
+    },
+    per_vu_iters: {
+      executor: 'per-vu-iterations',
+      exec: 'scenario2',
+      vus: 50,
+      iterations: 100,
+      startTime: '30s',
+      maxDuration: '1m',
+      tags: { scenario: '2' },
+      env: { SCENARIO: '2' },
```

Comment:
if we are going to be setting those in k6 it's probably not a good idea to give examples like that ;)

Reply:
Hah true, though I wrote this before the default `scenario` tag feature, and well, because of lack of imagination :)

I changed it slightly in f99fbf3, but let me know if you have a better idea.